sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -300,5 +287,107 @@ private static void AddTelemetryFromActivityEvents(Activity activity, TelemetryI
                 BaseData = exceptionData,
             };
         }
+
+        internal static void AddEnqueuedTimeToMeasurrementsAndLinksToProperties(Activity activity, IDictionary<string, double> measurements, ref AzMonList UnMappedTags)
+        {
+            if (activity.Links != null && activity.Links.Any())
+            {
+                if (TryGetAverageQueueTimeWithLinks(activity, ref UnMappedTags, out long enqueuedTime))
+                {
+                    measurements["timeSinceEnqueued"] = enqueuedTime;
+                }
+            }
+        }
+
+        private static bool TryGetAverageQueueTimeWithLinks(Activity activity, ref AzMonList UnMappedTags, out long avgTimeInQueue)
+        {
+            avgTimeInQueue = 0;
+            var linksCount = 0;
+            DateTimeOffset startTime = activity.StartTimeUtc;
+            long startEpochTime = startTime.ToUnixTimeMilliseconds();
+            bool isEnqueuedTimeCalculated = true;
+
+            string msLinks = "_MS.links";
+            var linksJson = new StringBuilder();
+            linksJson.Append('[');
+            foreach (ref readonly var link in activity.EnumerateLinks())
+            {
+                long msgEnqueuedTime = 0;
+                if (isEnqueuedTimeCalculated && !TryGetEnqueuedTime(link, out msgEnqueuedTime))
```

Comment:
nit: For better readability. 
```suggestion
               if (isEnqueuedTimeCalculated)
                {
                 if(!TryGetEnqueuedTime(link, out msgEnqueuedTime))
                 {
                    // instrumentation does not consistently report enqueued time, ignoring whole span
                     isEnqueuedTimeCalculated = false;
                    }
                    avgTimeInQueue += Math.Max(startEpochTime - msgEnqueuedTime, 0);
                }
```